include/operations/blas_operators.hpp

```diff
@@ -40,22 +41,19 @@ namespace blas {
 template <typename ScalarT>
 struct IndVal {
   using value_type = ScalarT;
-  size_t ind;
-  value_type val;
+  size_t ind = UINT_MAX;
+  value_type val = 0.;
```

Comment:
Is there any reason that you have assign the value here instead of using the constructor?